src/pageObjects/components/RegistrationPopup.js

```diff
@@ -0,0 +1,58 @@
+import BaseComponent from "../BaseComponent.js";
+import { expect } from "@playwright/test";
+
+export default class RegistrationPopup extends BaseComponent {
+  nameInput;
+  lastNameInput;
+  emailInput;
+  passwordInput;
+  repeatPasswordInput;
+  registerButton;
+  actualErrorsText = [];
+  errors;
+
+  constructor(page) {
+    super(page, page.locator("div.modal-content"));
+    this.page = page;
+    this.nameInput = page.locator("input#signupName");
+    this.lastNameInput = page.locator("input#signupLastName");
+    this.emailInput = page.locator("input#signupEmail");
+    this.passwordInput = page.locator("input#signupPassword");
+    this.repeatPasswordInput = page.locator("input#signupRepeatPassword");
+    this.registerButton = page.locator("button.btn.btn-primary:nth-child(1)");
+    this.errors = this.page.locator("form div > p");
+  }
+
+  async fillOutFields(name, lastName, email, password, repeatPassword) {
+    await this.nameInput.fill(name);
+    await this.lastNameInput.fill(lastName);
+    await this.emailInput.fill(email);
+    await this.passwordInput.fill(password);
+    await this.repeatPasswordInput.fill(repeatPassword);
+    await this.repeatPasswordInput.blur();
+  }
+
+  async registerUser() {
+    await this.registerButton.click();
+    await expect(this.page).toHaveURL("/panel/garage");
+  }
+
+  async clickFields() {
+    await this.nameInput.click();
+    await this.lastNameInput.click();
+    await this.emailInput.click();
+    await this.passwordInput.click();
+    await this.repeatPasswordInput.click();
+    await this.repeatPasswordInput.blur();
+  }
+  async getErrorsText() {
+    for (const errorItem of await this.errors.all()) {
+      const text = await errorItem.innerText();
+      this.actualErrorsText.push(text);
+    }
+    return this.actualErrorsText;
```

Comment:
do not store as PO property, create new array right here and return it as result